socorro/external/es/query.py

```diff
@@ -7,33 +7,26 @@
 import json
 import re
 
-from configman import RequiredConfig, Namespace, class_converter
-
 from socorro.lib import DatabaseError, MissingArgumentError, ResourceNotFound
 from socorro.external.es.base import generate_list_of_indexes
 from socorro.external.es.supersearch import BAD_INDEX_REGEX
 from socorro.lib import libdatetime, external_common
 
 
-class Query(RequiredConfig):
+class Query:
     """Implement the /query service with ElasticSearch."""
 
-    required_config = Namespace()
-    required_config.add_option(
-        "elasticsearch_class",
-        doc="a class that implements the ES connection object",
-        default="socorro.external.es.connection_context.ConnectionContext",
-        from_string_converter=class_converter,
-    )
-
     filters = [("query", None, "json"), ("indices", None, ["list", "str"])]
 
-    def __init__(self, config):
-        self.config = config
-        self.context = self.config.elasticsearch_class(self.config)
+    def __init__(self, crashstorage, timeout=120):
```

Comment:
I like this change a lot. Instead of query accessing the connection context directly, it now goes through a crashstorage. That makes tons of sense since the crashstorage defines all the structures the query is querying.

This is true for supersearch, too.